pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -155,6 +155,12 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) && !localFeatureGate.Enabled(features.GracefulNodeShutdown) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: Specifying ShutdownGracePeriod or ShutdownGracePeriodCriticalPods requires feature gate GracefulNodeShutdown"))
 	}
+	if localFeatureGate.Enabled(features.PodPriorityBasedGracefulShutdown) && len(kc.PodPriorityShutdownGracePeriods) != 0 && (kc.ShutdownGracePeriod.Duration > 0 || kc.ShutdownGracePeriodCriticalPods.Duration > 0) {
```

Comment:
I would also expect a validation rule that fails if you specify both pod priority and critical pod shutdown grace period (and the comment in the API doc explaining that).